Improve trainer back pic-related data and code (+ smol documentation) (#6821)
rh-hideout/pokeemerald-expansion: docs/tutorials/how_to_trainer_back_pic.md

# How to add a new trainer back pic

## Content
* [Quick Summary](#quick-summary)
* [The Graphics](#the-graphics)
  * [1. Edit the sprites](#2-edit-the-sprites)
  * [2. Register the sprites](#2-register-the-sprites)
  * [3. Connecting pictures to the data](#2-connecting-pictures-to-the-data)
* [The Data](#the-data)
  * [4. Defining the trainer back pic](#2-defining-the-trainer-back-pic)
* [Usage](#usage)

## Quick Summary
If you've done this before and just need a quick lookup, here's what files you need:

> 1. GFX into [`graphics/trainers/back_pics`](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/graphics/trainers/back_pics).

Reviewer: ```suggestion
1. Place graphics in [`graphics/trainers/back_pics`](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/graphics/trainers/back_pics).
```

I'm 99% certain these links can (and should) be updated to be relational links. With that change, people can access while offline or if `master` changes names or if GitHub changes their URL structure.

Author: if I understand this correctly, I just need to change any `https://github.com/rh-hideout/pokeemerald-expansion/blob/master/<file path>` to something like `./<file path>` right? or just `/<file path>`

2. Point game to where graphic files are found: [`src/data/graphics/trainers`](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/src/data/graphics/trainers.h).
3. Add trainer to [`include/constants/trainers.h`](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/include/constants/trainers.h),

## The Graphics

### 1. Add the sprites

> We will start with a graphic that we want to use for our new trainer pic. Unlike with adding Pokémon, the trainer sprites aren't sorted in individual folders, but rather in one folder: [`graphics/trainers/back_pics`](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/graphics/trainers/back_pics).
> 
> **Remember to limit yourself to 16 colors including transparency in the first slot!**

Reviewer: ```suggestion
We will start with a graphic that we want to use for our new trainer pic. Unlike with adding Pokémon, the trainer sprites aren't sorted in individual folders, but rather in one folder: [`graphics/trainers/back_pics`](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/graphics/trainers/back_pics). **Trainers sprites cannot be more than 16 - this includes the color that will be transparent, which is the first slot of the palette.**
```

So, finally, it needs to look like this:
```diff
 #define TRAINER_BACK_SPRITE(trainerPic, yOffset, sprite, pal, anim)                          \
     [trainerPic] =                                                                           \
     {                                                                                        \
         .coordinates = {.size = 8, .y_offset = yOffset},                                     \
         .backPic = {.data = sprite, .size = TRAINER_PIC_SIZE, .relativeFrames = TRUE},       \
         .palette = {.data = pal, .tag = trainerPic},                                         \
         .animation = anim,                                                                   \
     }

 const struct TrainerBacksprite gTrainerBacksprites[] =
 {
     TRAINER_BACK_SPRITE(TRAINER_BACK_PIC_BRENDAN, 4, gTrainerBackPic_Brendan, gTrainerPalette_Brendan, sBackAnims_Hoenn),
     ...
     TRAINER_BACK_SPRITE(TRAINER_BACK_PIC_STEVEN, 4, gTrainerBackPic_Steven, gTrainerPalette_Steven, sBackAnims_Hoenn),
+    TRAINER_BACK_SPRITE(TRAINER_BACK_PIC_NEW_ONE, 4, gTrainerBackPic_NewOne, gTrainerBackPicPalette_NewOne, sBackAnims_Hoenn),
};
```

> Note: If you have your back pic structured like FRLG Red (which uses 5 frames instead of 4), change the `4` to `5` and the `sBackAnims_Hoenn` to `sBackAnims_Kanto`.

Reviewer: ```suggestion
**Note**: Trainer back pics can have 4 or 5 frames of animation. Trainers with 5 frames must have their `yOffset` set to 5, and their `anim` set to `sBackAnims_Kanto`.
```

### The Data
#### 4. Defining the trainer back pic
Finally, let's bring it all together by defining our new trainer pic in [`include/constants/trainers.h`](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/include/constants/trainers.h):

```diff
 #define TRAINER_BACK_PIC_WALLY                  6
 #define TRAINER_BACK_PIC_STEVEN                 7
+#define TRAINER_BACK_PIC_NEW_ONE                8
```

> Remember to count the number next to the trainer pic up by one!

Reviewer: Curious what @AlexOn1ine thinks, but is there any reason this can't be an enum?

## Usage

> You can test your new trainer back pic by going to [`src/data/battle_partners.party`](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/src/data/battle_partners.party) and change the `Pic` field. For example:

Reviewer: ```suggestion
You can test your new trainer back pic by going to [`src/data/battle_partners.party`](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/src/data/battle_partners.party) and change the `Pic` field. The syntax should match the constant (`TRAINER_BACK_PIC_NEW_ONE`) with the underscore replaced by spaces. For example:
```
